src/e2e/java/teammates/e2e/util/BackDoor.java

```diff
@@ -212,12 +213,14 @@ private static void addAuthKeys(HttpRequestBase request) {
      * However, removing the data bundle on teardown manually is not a perfect solution because two tests can concurrently
      * access the same account and their data may get mixed up in the process. This is a major problem we need to address.
      */
-    public static String removeAndRestoreDataBundle(DataBundle dataBundle) {
+    public static String removeAndRestoreDataBundle(DataBundle dataBundle) throws HttpRequestFailedException {
         removeDataBundle(dataBundle);
         ResponseBodyAndCode putRequestOutput =
                 executePostRequest(Const.ResourceURIs.DATABUNDLE, null, JsonUtils.toJson(dataBundle));
-        return putRequestOutput.responseCode == HttpStatus.SC_OK
-                ? Const.StatusCodes.BACKDOOR_STATUS_SUCCESS : Const.StatusCodes.BACKDOOR_STATUS_FAILURE;
+        if (putRequestOutput.responseCode == HttpStatus.SC_OK) {
+            return putRequestOutput.responseBody;
+        }
+        throw new HttpRequestFailedException("Request failed with status code: " + putRequestOutput.responseCode);
```

Comment:
It OK to use `IOException` I think.

Reply:
I felt that Http requests should use a separate exception because they aren't exactly IO, and you might want to handle them separately if you need to.